CI unit tests fail in forked repositories

The CI unit test from #67 is failing for PRs from forked repositories (see second-last line).

```
2025-09-20T12:35:33.9186684Z ##[group]Run dorny/test-reporter@v1
2025-09-20T12:35:33.9186946Z with:
2025-09-20T12:35:33.9187122Z   name: JUnit Tests
2025-09-20T12:35:33.9187324Z   path: test_results/test_results.xml
2025-09-20T12:35:33.9187573Z   reporter: java-junit
2025-09-20T12:35:33.9187784Z   path-replace-backslashes: false
2025-09-20T12:35:33.9188018Z   list-suites: all
2025-09-20T12:35:33.9188195Z   list-tests: all
2025-09-20T12:35:33.9188381Z   max-annotations: 10
2025-09-20T12:35:33.9188570Z   fail-on-error: true
2025-09-20T12:35:33.9188916Z   fail-on-empty: true
2025-09-20T12:35:33.9189106Z   only-summary: false
2025-09-20T12:35:33.9189420Z   token: ***
2025-09-20T12:35:33.9189591Z env:
2025-09-20T12:35:33.9189821Z   pythonLocation: /opt/hostedtoolcache/Python/3.11.13/x64
2025-09-20T12:35:33.9190189Z   LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.13/x64/lib
2025-09-20T12:35:33.9190472Z ##[endgroup]
2025-09-20T12:35:34.0390313Z Action was triggered by pull_request: using SHA from head of source branch
2025-09-20T12:35:34.0401471Z Check runs will be created with SHA=091c88b03e986f4704cff1090ed8e2781267c24f
2025-09-20T12:35:34.0405615Z ##[group]Listing all files tracked by git
2025-09-20T12:35:34.0448383Z [command]/usr/bin/git ls-files -z
2025-09-20T12:35:34.0517222Z .env�.github/workflows/build_test.yml�.gitignore�.travis.yml�LICENSE�README.md�examples/__init__.py�examples/dis_receiver.py�examples/dis_sender.py�opendis/DataInputStream.py�opendis/DataOutputStream.py�opendis/PduFactory.py�opendis/RangeCoordinates.py�opendis/__init__.py�opendis/dis7.py�opendis/stream.py�opendis/types.py�poetry.lock�pyproject.toml�tests/ElectromagneticEmissionPdu-single-system.raw�tests/EntityStatePdu-26.raw�tests/SetDataPdu-multi-variable-datums.raw�tests/SetDataPdu-vbs-script-cmd.raw�tests/SignalPdu.raw�tests/TransmitterPdu.raw�tests/__init__.py�tests/test_ElectromageneticEmissionPdu.py�tests/test_EntityStatePdu.py�tests/test_RangeCoordinates.py�tests/test_SetDataPdu.py�tests/test_SignalPdu.py�tests/test_TransmitterPdu.py�tests/test_examples.py�
2025-09-20T12:35:34.0521208Z ##[endgroup]
2025-09-20T12:35:34.0521461Z Found 33 files tracked by GitHub
2025-09-20T12:35:34.0521817Z Using test report parser 'java-junit'
2025-09-20T12:35:34.0579148Z ##[group]Creating test report JUnit Tests
2025-09-20T12:35:34.0580928Z Processing test results for check run JUnit Tests
2025-09-20T12:35:34.0621235Z Creating check run JUnit Tests
2025-09-20T12:35:34.2494178Z ##[endgroup]
2025-09-20T12:35:34.2511521Z ##[error]HttpError: Resource not accessible by integration
2025-09-20T12:35:34.3337181Z Post job cleanup.
```

The [documentation for the `dorny/test-reporter` GitHub Action](https://github.com/dorny/test-reporter?tab=readme-ov-file#recommended-setup-for-public-repositories) explains how to resolve it:

> Workflows triggered by pull requests from forked repositories are executed with read-only token and therefore can't create check runs. To workaround this security restriction, it's required to use two separate workflows

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,7 +1,7 @@
 # This workflow will install opendis dependencies, and then run tests and lint with a variety of Python versions
 # For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
 
-name: Build and Test OpenDis
+name: CI
 
 on:
   push:
@@ -9,7 +9,7 @@ on:
   pull_request:
 
 jobs:
-  build:
+  build-test:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
@@ -28,16 +28,10 @@ jobs:
         poetry install --with dev
     - name: Test with pytest
       run: |
-        poetry run pytest --junitxml=./test_results/test_results.xml --cov=opendis --cov-report term --cov-report=xml:./test_results/coverage.xml
-    - name: Upload Test Results
-      uses: dorny/test-reporter@v1
-      if: always()
+        poetry run pytest --junitxml=./test_results/test_results_${{ matrix.python-version }}.xml --cov=opendis --cov-report term --cov-report=xml:./test_results/coverage_${{ matrix.python-version }}.xml
+    - name: Upload test results
+      uses: actions/upload-artifact@v4
+      if: ${{ !cancelled() }}
       with:
-        name: JUnit Tests
-        path: test_results/test_results.xml
-        reporter: java-junit
-    - name: Pytest coverage comment
-      uses: MishaKav/pytest-coverage-comment@main
-      with:
-        pytest-xml-coverage-path: ./test_results/coverage.xml
-        junitxml-path: ./test_results/test_results.xml
+        name: test-results-${{ matrix.python-version }}
+        path: test_results/

--- a/.github/workflows/test-report.yml
+++ b/.github/workflows/test-report.yml
@@ -1,0 +1,44 @@
+name: 'Test Report'
+on:
+  workflow_run:
+    workflows: ['CI']                     # runs after CI workflow
+    types:
+      - completed
+permissions:
+  contents: read
+  actions: read
+  checks: write
+jobs:
+  report:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: dorny/test-reporter@v2
+      with:
+        artifact: test-results-3.10       # artifact name for Python 3.10
+        name: Python 3.10 Tests          # Name of the check run which will be created
+        path: '*.xml'                     # Path to test results (inside artifact .zip)
+        reporter: java-junit              # Format of test results
+    - uses: dorny/test-reporter@v2
+      with:
+        artifact: test-results-3.11       # artifact name for Python 3.11
+        name: Python 3.11 Tests          # Name of the check run which will be created
+        path: '*.xml'                     # Path to test results (inside artifact .zip)
+        reporter: java-junit              # Format of test results
+  
+  coverage-report:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/download-artifact@v4
+      with:
+        name: test-results-3.10
+        path: test_results_3_10/
+    - uses: actions/download-artifact@v4
+      with:
+        name: test-results-3.11
+        path: test_results_3_11/
+    - name: Pytest coverage comment
+      uses: MishaKav/pytest-coverage-comment@main
+      with:
+        pytest-xml-coverage-path: test_results_3_10/coverage_3.10.xml
+        junitxml-path: test_results_3_10/test_results_3.10.xml

